plugins/woocommerce-blocks/assets/js/blocks/filter-wrapper/edit.tsx

```diff
@@ -1,19 +1,100 @@
 /**
  * External dependencies
  */
-import { useBlockProps, InnerBlocks } from '@wordpress/block-editor';
-import type { BlockEditProps } from '@wordpress/blocks';
+import {
+	useBlockProps,
+	InnerBlocks,
+	InspectorControls,
+	BlockControls,
+} from '@wordpress/block-editor';
+import { createBlock, type BlockEditProps } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import { replace } from '@wordpress/icons';
+import {
+	PanelBody,
+	Button,
+	ToolbarButton,
+	ToolbarGroup,
+} from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import { Attributes } from './types';
 
-const Edit = ( { attributes }: BlockEditProps< Attributes > ) => {
+const Edit = ( { attributes, clientId }: BlockEditProps< Attributes > ) => {
+	const block = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getBlock( clientId );
+	} );
+	const { replaceBlock } = useDispatch( 'core/block-editor' );
+
+	const upgradeBlock = useCallback( () => {
```

Comment:
You'll need to gate this the same way we do the new experimental blocks because if the new blocks are not registered this button will throw an error.

plus we are about to do a mass rename/consolidation so this code should not be present in production yet anyway, I don't think?

Reply:
Good catch! I https://github.com/woocommerce/woocommerce/pull/43218/commits/1cfb7826c9d9cd28a78da0279619bd4ffc79ee5e the logic into the upgrade component and add a guard before rendering.